packages/unstake/src/broker.rs

```diff
@@ -76,14 +76,14 @@ impl Broker {
         let max_rate = rates.vault_max_interest;
 
         // Calculate the value of the Unstaked amount, in terms of the underlying asset. I.e. the max amount we'll need to borrow
-        let value = unbond_amount.mul(rates.provider_redemption);
+        let value = unbond_amount.mul_floor(rates.provider_redemption);
 
         // For now we'll naively assume that the borrow rate will stay fixed for the duration of the unbond.
         // During periods of high interest, Unstakes will cost more and a user will have to wait for the rate
         // to fall if they want a more favourable rate.
         let offer_rate = max(current_rate, self.min_rate);
         let max_rate_shortfall = max_rate
-            .checked_sub(current_rate)
+            .checked_sub(offer_rate)
```

Comment:
Pretty sure this should be `offer_rate`, since we may be able to guarantee a higher than `current_rate` fee